SonarQube.TeamBuild.PostProcessor/SummaryReportBuilder.cs

```diff
@@ -0,0 +1,164 @@
+﻿using SonarQube.Common;
+using SonarQube.TeamBuild.Integration;
+using SonarRunner.Shim;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+
+namespace SonarQube.TeamBuild.PostProcessor
+{
+    /// <summary>
+    /// Generates summary reports for various build systems
+    /// </summary>
+    internal class SummaryReportBuilder
+    {
+        private AnalysisConfig config;
+        private ILogger logger;
+        private ProjectInfoAnalysisResult result;
+        private TeamBuildSettings settings;
+
+        private SummaryReportBuilder(TeamBuildSettings settings, AnalysisConfig config, ProjectInfoAnalysisResult result, ILogger logger)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+            if (config == null)
+            {
+                throw new ArgumentNullException(nameof(config));
+            }
+            if (result == null)
+            {
+                throw new ArgumentNullException(nameof(result));
+            }
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            this.settings = settings;
+            this.config = config;
+            this.result = result;
+            this.logger = logger;
+        }
+
+        /// <summary>
+        /// Generates summary reports for LegacyTeamBuild and for Build Vnext
+        /// </summary>
+        public static void GenerateReports(TeamBuildSettings settings, AnalysisConfig config, ProjectInfoAnalysisResult result, ILogger logger)
+        {
+            SummaryReportBuilder reportBuilder = new SummaryReportBuilder(settings, config, result, logger);
+            reportBuilder.GenerateReports();
+        }
+
+        private void GenerateReports()
+        {
+            if (this.settings.BuildEnvironment == BuildEnvironment.LegacyTeamBuild
+                && !TeamBuildSettings.SkipLegacyCodeCoverageProcessing)
+            {
+                UpdateTeamBuildSummary();
+            }
+
+            if (this.settings.BuildEnvironment == BuildEnvironment.TeamBuild)
+            {
+                CreateBuildVnextSummaryMdFile();
+            }
+        }
+
+        private void CreateBuildVnextSummaryMdFile()
+        {
```

Comment:
When we discussed this earlier, I thought we'd concluded that there was no reason not to write the MD file in all cases, regardless of the build system. Did you find a reason to make this Build vNext only?